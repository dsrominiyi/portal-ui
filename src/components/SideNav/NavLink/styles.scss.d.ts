export const navLink: string;
export const active: string;
export const mini: string;
