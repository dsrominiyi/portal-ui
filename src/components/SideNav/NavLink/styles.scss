@import '../../../style/_vars';

a.nav-link {
  display: flex;
  align-items: center;
  padding: 1rem;
  color: $brand-white;
  font-size: 0.85rem;
  cursor: pointer;
  text-decoration: none;

  &:hover,
  &.active,
  &.active:hover {
    color: $brand-light-green;
  }

  i {
    margin-right: 0.5rem;
  }

  &.mini {
    flex-direction: column;
    font-size: 0.65rem;
    text-align: center;

    i {
      margin: 0 0 0.25rem 0;
    }
  }
}
