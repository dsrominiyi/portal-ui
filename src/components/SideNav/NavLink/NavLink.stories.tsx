import React, { ReactNode } from 'react';
import { storiesOf } from '@storybook/react';

import NavLink from '.';

import styles from '../../../../.storybook/styles.scss';

const stories = storiesOf('NavLink', module);

const onClickDefault = () => alert('You clicked!');

const render = (
  label: string,
  icon: ReactNode,
  href?: string,
  isActive?: boolean,
  isMiniLink?: boolean,
  onClick: () => void = onClickDefault
) => () => (
  <NavLink
    label={label}
    icon={icon}
    href={href}
    isActive={isActive}
    isMiniLink={isMiniLink}
    onClick={href ? undefined : onClick}
  />
);

const renderWithBackground = (renderFunc: () => ReactNode, mini?: boolean) => () => (
  <div className={styles.darkBackground} style={{ width: mini ? 105 : 224 }}>
    {renderFunc()}
  </div>
);

export const storyRenders = {
  withHref: render('Customer Overview', <i className="fas fa-user" />, 'https://www.google.com'),
  withOnClick: render('Policies', <i className="fas fa-folder" />),
  active: render('Diary', <i className="fas fa-calendar-alt" />, undefined, true),
  mini: render('Customer Overview', <i className="fas fa-user" />, undefined, false, true)
};

stories.add('With href', renderWithBackground(storyRenders.withHref));
stories.add('With onClick function', renderWithBackground(storyRenders.withOnClick));
stories.add('With active state', renderWithBackground(storyRenders.active));
stories.add('Mini Link', renderWithBackground(storyRenders.mini, true));
