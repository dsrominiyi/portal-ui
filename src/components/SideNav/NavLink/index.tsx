import React, { FunctionComponent, ReactNode, MouseEvent } from 'react';
import classnames from 'classnames/bind';

import Link from '../../Link';

import styles from './styles.scss';

interface Props {
  label: string;
  icon: ReactNode;
  href?: string;
  isActive?: boolean;
  isMiniLink?: boolean;
  onClick?(event: MouseEvent): void;
}

const cx = classnames.bind(styles);

export const NavLink: FunctionComponent<Props> = ({
  label,
  icon,
  href,
  isActive,
  isMiniLink,
  onClick
}: Props) => (
  <Link
    classNames={[
      cx({
        navLink: true,
        active: isActive,
        mini: isMiniLink
      })
    ]}
    label={label}
    icon={icon}
    href={href}
    onClick={onClick}
  />
);

NavLink.displayName = 'NavLink';
export default NavLink;
