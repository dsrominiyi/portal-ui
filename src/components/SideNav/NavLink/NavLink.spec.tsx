import { storyRenders } from './NavLink.stories';

import testStorySnapshots from '../../../helpers/test/testStorySnapshots';

describe('NavLink', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
