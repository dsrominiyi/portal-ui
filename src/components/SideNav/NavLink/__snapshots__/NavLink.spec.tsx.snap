// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component story renders match snapshots: active 1`] = `
<Link
  classNames={
    Array [
      "navLink active",
    ]
  }
  icon={
    <i
      className="fas fa-calendar-alt"
    />
  }
  label="Diary"
  onClick={[Function]}
/>
`;

exports[`NavLink component story renders match snapshots: mini 1`] = `
<Link
  classNames={
    Array [
      "navLink mini",
    ]
  }
  icon={
    <i
      className="fas fa-user"
    />
  }
  label="Customer Overview"
  onClick={[Function]}
/>
`;

exports[`NavLink component story renders match snapshots: withHref 1`] = `
<Link
  classNames={
    Array [
      "navLink",
    ]
  }
  href="https://www.google.com"
  icon={
    <i
      className="fas fa-user"
    />
  }
  label="Customer Overview"
/>
`;

exports[`NavLink component story renders match snapshots: withOnClick 1`] = `
<Link
  classNames={
    Array [
      "navLink",
    ]
  }
  icon={
    <i
      className="fas fa-folder"
    />
  }
  label="Policies"
  onClick={[Function]}
/>
`;
